### PR TITLE
Added documentation to embed component files.

### PR DESCRIPTION
**Pull request by Red-K0**

### CODE REVIEW

**Comment by Red-K0** [comment]
Took way less time, given how repetitive some of the files are.

File: NetCord/EmbedField.cs [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
 
+    /// <summary>
+    /// Equivalent to <see cref="Embed.Title"/> but localised to a field, limited to 256 characters.
+    /// </summary>
     public string Name => _jsonModel.Name;
+
+    /// <summary>
+    /// Equivalent to <see cref="Embed.Description"/> but localised to a field, limited to 1024 characters.
+    /// </summary>
+
```
Comment (KubaZ2) [inline NetCord/EmbedField.cs]:
```suggestion
```

File: NetCord/EmbedField.cs [hunk 2]
```diff
@@ -1,16 +1,29 @@
...
+    /// Equivalent to <see cref="Embed.Title"/> but localised to a field, limited to 256 characters.
+    /// </summary>
     public string Name => _jsonModel.Name;
+
+    /// <summary>
+    /// Equivalent to <see cref="Embed.Description"/> but localised to a field, limited to 1024 characters.
+    /// </summary>
+    
     public string Value => _jsonModel.Value;
+    /// <summary>
```
Comment (KubaZ2) [inline NetCord/EmbedField.cs:25]:
```suggestion

    /// <summary>
```

File: NetCord/EmbedProvider.cs [hunk 3]
```diff
@@ -1,15 +1,24 @@
...
         _jsonModel = jsonModel;
     }
 
+    /// <summary>
+    /// The name of the provider, displayed at the top of the embed.
+    /// </summary>
     public string? Name => _jsonModel.Name;
+
+    /// <summary>
+    /// When set, turns the <see cref="Name"/> into a clickable link, pointing to the specified URL.
```
Comment (KubaZ2) [inline NetCord/EmbedProvider.cs]: It points to the scheme, subdomain, and domain, the scheme is `https` when `http` is provided though, the path is omitted

**Review by KubaZ2 — CHANGES_REQUESTED** [review]
Also `discord` should be changed to `Discord`

File: NetCord/EmbedProvider.cs (hunk 3 above)
Comment (Red-K0) [reply NetCord/EmbedProvider.cs]: How so?

Comment (KubaZ2) [reply NetCord/EmbedProvider.cs]: What about this?
```suggestion
    /// When set, turns the <see cref="Name"/> into a clickable link, pointing to the base of the specified URL.
```

**Comment by Red-K0** [comment]
> Also `discord` should be changed to `Discord`

All instances changed

File: NetCord/EmbedProvider.cs (hunk 3 above)
Comment (Red-K0) [reply NetCord/EmbedProvider.cs]: Ah I get what you mean now

**Comment by Red-K0** [comment]
URL references also changed

File: NetCord/EmbedAuthor.cs [hunk 4]
```diff
@@ -18,7 +18,7 @@ public EmbedAuthor(JsonModels.JsonEmbedAuthor jsonModel)
     public string? Name => _jsonModel.Name;
     
     /// <summary>
-    /// When set, turns the <see cref="Name"/> into a clickable link, pointing to the specified URL.
+    /// When set, turns the <see cref="Name"/> into a clickable link, pointing to the base of the specified URL.
```
Comment (KubaZ2) [inline NetCord/EmbedAuthor.cs]: The URL doesn't point to the base of the specified URL. For example for youtube it points to the video author. I think the previous wording was better.

**Review by KubaZ2 — CHANGES_REQUESTED** [review]

File: NetCord/EmbedAuthor.cs (hunk 4 above)
Comment (Red-K0) [reply NetCord/EmbedAuthor.cs]: Will revert then

Comment (Red-K0) [reply NetCord/EmbedAuthor.cs]: Done and done

File: NetCord/EmbedProvider.cs (hunk 3 above)
Comment (KubaZ2) [inline NetCord/EmbedProvider.cs]: You have reverted a wrong line

Comment (Red-K0) [reply NetCord/EmbedProvider.cs]: Reverted the revert and reverted the correct one (sorry about that lol)

**Comment by KubaZ2** [comment]
Thanks

**Pull request merged**